airflow/providers/amazon/aws/transfers/redshift_to_s3.py

```
@@ -108,8 +115,15 @@ def execute(self, context) -> None:
 
         credentials = s3_hook.get_credentials()
         unload_options = '\n\t\t\t'.join(self.unload_options)
-        s3_key = f'{self.s3_key}/{self.table}_' if self.table_as_file_name else self.s3_key
-        select_query = f"SELECT * FROM {self.schema}.{self.table}"
+        s3_key = f'{self.s3_key}/{self.table}_' if (self.table and self.table_as_file_name) else self.s3_key
+        if self.select_query:
+            select_query = self.select_query
+        elif self.schema and self.table:
+            select_query = f"SELECT * FROM {self.schema}.{self.table}"
+        else:
+            self.log.info('Please provide both `schema` and `table` params to fetch the data.')
+            return
```

You can move this to `__init__`  and set `self._select_query` then it is easy to test too

i.e. just pass schema and table and see what `self._select_query` gives you and do the same with just passing `select_query`

---

Yeah, testing will be easy after that, will do.

Will it be a good idea to throw an error from `__init__` or add a check in `execute` for `self._select_query` (`else` case scenario)?

---

It will good to fail early -- so throwing an error from `__init__` would be ideal

---

got it!